### PR TITLE
feat: show metadata in run table

### PR DESCRIPTION
**Pull request by keita-determined**
## Ticket

[ET-222]
[ET-223]

## Description

Show metadata in the run table.
have not excluded the array type columns yet. will implement it after #9759 is merged, but its not a blocker to merge this PR.

## Test Plan

- Go to run table
- Click `Columns` button and see `Metadata` tab. user should be able to add metadata columns and filter and sort by metadata (metadata with array type are not supported for filtering and sorting, PR will come after #9759 to exclude the array type metadata in filter and sort)

## Checklist

- [x] Changes have been manually QA'd
- [ ] New features have been approved by the corresponding PM
- [ ] User-facing API changes have the "User-facing API Change" label
- [ ] Release notes have been added as a separate file under `docs/release-notes/`
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses have been included for new code which was copied and/or modified from any external code

[ET-222]: https://hpe-aiatscale.atlassian.net/browse/ET-222?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[ET-223]: https://hpe-aiatscale.atlassian.net/browse/ET-223?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by EmilyBonar — CHANGES_REQUESTED** [review]
Screenshot 2024-08-01 at 14-03-45 Uncategorized Runs - Determined
This is really bad UX, needs some design work to figure out correct solution.

File: webui/react/src/pages/FlatRuns/FlatRuns.tsx [hunk 1]
```diff
@@ -364,6 +364,9 @@ const FlatRuns: React.FC<Props> = ({ projectId, workspaceId, searchId }) => {
           case V1LocationType.CUSTOMMETRIC:
             dataPath = `summaryMetrics.${currentColumn.column}`;
             break;
+          case V1LocationType.RUNMETADATA:
```
[the remark this answers is not in the document]
Comment (keita-determined) [reply webui/react/src/pages/FlatRuns/FlatRuns.tsx]: fixed it

**Comment by keita-determined** [comment]
@EmilyBonar 
image

fixed it

File: webui/react/src/components/ColumnPickerMenu.module.scss [hunk 2]
```diff
@@ -18,3 +18,9 @@
     white-space: nowrap;
   }
 }
+
+@media only screen and (width <= 390px) {
```
Comment (keita-determined) [inline webui/react/src/components/ColumnPickerMenu.module.scss:22]: looks like `$breakpoint-` variables are not working so use the constant number

**Review by EmilyBonar — APPROVED** [review]
LGTM

**Pull request merged**